src/Provider/Clock.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace League\OAuth2\Client\Provider;
+
+/**
+ * Represents an implementation of a Clock.
+ */
+class Clock
```

Comment:
Is there a reason not to have something like `ClockInterface` here to require the `now()` method?

Reply:
Im all for it but interfaces seem to be used sparingly on this project... I wish providers had an interface.

Reply:
I think an interface is appropriate here, because the implementation of `ClockInterface` should be `final class ...`.